app/src/main/java/com/smashingboxes/epa_prototype_android/MainActivityAdapter.java

```diff
@@ -0,0 +1,368 @@
+import com.h6ah4i.android.widget.advrecyclerview.swipeable.action.SwipeResultActionMoveToSwipedDirection;
+import com.h6ah4i.android.widget.advrecyclerview.utils.AbstractSwipeableItemViewHolder;
+import com.h6ah4i.android.widget.advrecyclerview.utils.RecyclerViewAdapterUtils;
+import com.smashingboxes.epa_prototype_android.fitbit.models.TimeSeries;
+import com.smashingboxes.epa_prototype_android.helpers.Utils;
+import com.smashingboxes.epa_prototype_android.network.epa.models.AirQuality;
+import com.smashingboxes.epa_prototype_android.network.epa.models.EpaActivity;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class MainActivityAdapter extends RecyclerView.Adapter<MainActivityAdapter.ViewHolder> implements SwipeableItemAdapter<MainActivityAdapter.ViewHolder> {
+
+    public static class ViewHolder extends AbstractSwipeableItemViewHolder {
+
+        View swipeContainer;
+        View airQualityBorder;
+        TextView activityDateText;
+        TextView activityTypeText;
+        TextView activityDistanceText;
+        TextView activityDistanceUnit;
+        TextView activityLocation;
+        TextView activityOutside;
+        TextView activityInside;
+
+        public ViewHolder(View view) {
+            super(view);
+            swipeContainer = view.findViewById(R.id.swipe_container);
+            airQualityBorder = view.findViewById(R.id.air_quality_status_border);
+            activityDateText = (TextView) view.findViewById(R.id.activity_date);
+            activityTypeText = (TextView) view.findViewById(R.id.activity_name);
+            activityDistanceText = (TextView) view.findViewById(R.id.activity_distance);
+            activityDistanceUnit = (TextView) view.findViewById(R.id.activity_unit);
+            activityLocation = (TextView) view.findViewById(R.id.activity_location);
+            activityOutside = (TextView) view.findViewById(R.id.activity_outside);
+            activityInside = (TextView) view.findViewById(R.id.activity_inside);
+        }
+
+        @Override
+        public View getSwipeableContainerView() {
+            return swipeContainer;
+        }
+
+    }
+
+    private MainActivity mainActivity;
+    private List<AirQuality> airQualities = new ArrayList<>();
+    private Map<String, TimeSeries> dateToActivityMap = new HashMap<>();
+    private AppStateManager appStateManager;
+    private Handler handler = new Handler();
+
+    public MainActivityAdapter(@NonNull MainActivity mainActivity) {
+        this.mainActivity = mainActivity;
+        this.appStateManager = AppStateManager.getInstance(mainActivity);
+        setHasStableIds(true);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return airQualities.get(position).getId();
+    }
+
+    public void queuedRemoveAll() {
+        for (final AirQuality airQuality : airQualities) {
```

Comment:
Would it be better to put the for loop within the new Runnable here, so that there aren't theoretically "lots" of new runnables created?
The same goes for the `queuedAddAll()` method
Although, I'm not sure what the trade-off would be.